### PR TITLE
Add manifests for vsphere paravirtual CPI.

### PR DESCRIPTION
**Pull request by HanFa**
## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Add manifests for vsphere paravirtual CPI with unit test. 
```

## Which issue(s) this PR fixes

Fixes: #

## Describe testing done for PR

Added four unit tests for paravirtual CPI objects.

### CODE REVIEW

**Comment by shyaamsn** [comment]
@blc1996 @vijaykatam PTAL

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 ---
 vsphereCPI:
+  mode: vsphereCPI
```
Comment (blc1996) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml:6]: Can we update the explanation of this field in README?

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml [hunk 2]
```diff
@@ -38,3 +39,9 @@ vsphereCPI:
   http_proxy: ""
   https_proxy: ""
   no_proxy: ""
+  clusterAPIVersion: "run.tanzu.vmware.com/v1alpha2"
+  clusterKind: "TanzuKubernetesCluster"
+  clusterName: "test-cluster-e2e-script"
```
Comment (blc1996) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml:44]: Why default the cluster name and UID to a specific value?

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.star [hunk 3]
```diff
@@ -14,6 +14,10 @@ def validate_vsphereCPI():
    end
 end
 
+def validate_vsphereParavirtualCPI():
```
Comment (blc1996) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.star:17]: Function with no contents?

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml (hunk 2 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml]: changed to empty values

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml (hunk 1 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml:6]: sure, updated those fields in readme

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.star (hunk 3 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.star:17]: validation has been added

**Comment by HanFa** [comment]
thanks @blc1996 
I have made some changes to address your comments. Including
```
update readme for new fields
add field validation in values.star
change default values to empty or something generic
```

Could you PTAL?

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    metadata:
+      labels:
+        name: guest-cluster-cloud-provider
+    spec:
+      hostNetwork: true
+      priorityClassName: system-cluster-critical
+      containers:
+        - args:
+            - --controllers=service
+            - --controllers=cloud-node
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml:25]: Where is this checking been taken care of?
https://gitlab.eng.vmware.com/core-build/cayman_cloud-provider-vsphere/-/blob/vmware-1.22.4/cloud_provider_vsphere/add-on/v1.22/cloud-controller-manager.yaml#L144-153

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            - --cloud-config=/config/cloud-config
+            #@yaml/text-templated-strings
+            - --cluster-name=(@= values.vsphereCPI.clusterName @)
+            - --cloud-provider=vsphere-paravirtual
+          env:
+            - name: SUPERVISOR_APISERVER_ENDPOINT_IP
+              value: #@ values.vsphereCPI.supervisorMasterEndpointIP
+            - name: SUPERVISOR_APISERVER_PORT
+              value: #@ values.vsphereCPI.supervisorMasterPort
+          image: localhost:5000/vmware.io/cloud-provider-vsphere-manager:v1.21.0_vmware.1
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml:35]: Image tag should be chnaged to 1.23

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml (hunk 4 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml:25]: added a TODO. will add a field `AntreaNSXRouted` once routable pods are supported

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml (hunk 5 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/upstream/vsphere-paravirtual-cpi/03-deployment.yaml]: fixed, good catch!

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md [hunk 6]
```diff
@@ -2,7 +2,8 @@
 
 > This package provides cloud provider interface using vsphere-cpi.
 
-For more information, see the [GitHub page](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere) of vSphere CPI.
+For more information, see the [GitHub page](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere) of vSphere
+CPI.
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md:6]: Un expected new line

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md [hunk 7]
```diff
@@ -41,6 +43,12 @@ None
 |`vsphereCPI.nsxt.rootCAData` | Optional | The certificate authority for the server certificate for locally signed certificates. Default: `""`. |
 | `vsphereCPI.nsxt.secretName` | Optional | The name of secret that stores CPI configuration. Default: `cloud-provider-vsphere-nsxt-credentials`. |
 | `vsphereCPI.nsxt.secretNamespace`| Optional | The namespace of secret that stores CPI configuration. Default: `True`. |
+| `vsphereCPI.clusterAPIVersion`| Optional for `vsphereParavirtual` | Used in `vsphereParavirtual` mode, defines the TKC API versions. Default: `run.tanzu.vmware.com/v1alpha2` |
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md:46]: Please replace TKC with Cluster

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.star [hunk 8]
```diff
@@ -14,6 +14,13 @@ def validate_vsphereCPI():
...
+   data.values.vsphereCPI.clusterName or assert.fail("vsphereParavirtualCPI TKC clusterName should be provided")
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.star:18]: Here also place replase TKC

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml [hunk 9]
```diff
@@ -38,3 +39,9 @@ vsphereCPI:
   http_proxy: ""
   https_proxy: ""
   no_proxy: ""
+  clusterAPIVersion: "run.tanzu.vmware.com/v1alpha2"
+  clusterKind: "TanzuKubernetesCluster"
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml:43]: Replace this with Cluster?

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go [hunk 10]
```diff
@@ -63,13 +82,111 @@ vsphereCPI:
...
+				values = defaultParavirtualValues
+			})
+
+			It("should render correct owner reference for the config map", func() {
+				configMaps := unmarshalConfigMaps(output)
+				Expect(configMaps).NotTo(BeEmpty())
+				ownerRefConfigMap := findConfigMapByName(configMaps, "ccm-owner-reference")
+				ownerRef, exists := ownerRefConfigMap.Data["owner-reference"]
+				Expect(exists).To(BeTrue())
+				Expect(ownerRef).To(Equal("{\"apiVersion\": \"run.tanzu.vmware.com/v1alpha2\",\n\"kind\": \"TanzuKubernetesCluster\",\n\"name\": \"tkg-cluster\",\n\"uid\": \"57341fa8-0983-472f-b744-00cf724dd307\"}\n"))
```
Comment (lubronzhan) [inline addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go:168]: Need to replace the TKC

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md (hunk 6 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/README.md:6]: fixed

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml (hunk 9 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/bundle/config/values.yaml:43]: replaced with `Cluster`

File: addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go (hunk 10 above)
Comment (HanFa) [reply addons/packages/vsphere-cpi/1.23.0-alpha.1/test/vsphere_cpi_test.go:168]: replaced with `Cluster`

**Review by lubronzhan — APPROVED** [review]
lgtm

**Pull request merged**